### PR TITLE
Add javadoc and source JAR plugins.

### PR DESCRIPTION
**Pull request by anwar6953**
This is needed to release hydrator-plugins 1.2.0.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -742,6 +743,50 @@
...
+          </plugin>
+
+          <!-- Javadoc jar -->
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>2.9.1</version>
+            <configuration>
+              <links>
+                <link>http://download.oracle.com/javase/6/docs/api/</link>
```
Comment (bdmogal) [inline pom.xml]: why do we use java 6? not 7 or 8? is it the same elsewhere?

**Comment by bdmogal** [comment]
1 question, other than that LGTM

**Comment by awholegunch** [comment]
In the CDAP pom.xml, this is configured to match a `jee.version`:

```
  <properties>
    <jee.version>7</jee.version>
...
<link>http://docs.oracle.com/javaee/${jee.version}/api/</link>
```

File: pom.xml (hunk 1 above)
Comment (anwar6953) [inline pom.xml]: Updated to use java 7. I had just copied this from kafka pack.

**Comment by awholegunch** [comment]
LGTM...

**Pull request merged**